common/src/KokkosFFT_padding.hpp

```diff
@@ -107,240 +105,128 @@ void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-
-  range_type range(
-      exec_space, point_type{{0, 0}}, point_type{{n0, n1}}, tile_type{{4, 4}}
-      // [TO DO] Choose optimal tile sizes for each device
-  );
-
-  Kokkos::parallel_for(
-      "KokkosFFT::crop_or_pad", range,
-      KOKKOS_LAMBDA(int i0, int i1) { out(i0, i1) = in(i0, i1); });
-}
-
-template <typename ExecutionSpace, typename InViewType, typename OutViewType>
-void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-                      OutViewType& out, shape_type<3> s) {
-  constexpr std::size_t DIM = 3;
-
-  auto [s0, s1, s2] = s;
-  out               = OutViewType("out", s0, s1, s2);
-
-  int n0 = std::min(s0, in.extent(0));
-  int n1 = std::min(s1, in.extent(1));
-  int n2 = std::min(s2, in.extent(2));
-
-  using range_type = Kokkos::MDRangePolicy<
-      ExecutionSpace,
-      Kokkos::Rank<DIM, Kokkos::Iterate::Default, Kokkos::Iterate::Default>>;
-  using tile_type  = typename range_type::tile_type;
-  using point_type = typename range_type::point_type;
-
-  range_type range(
-      exec_space, point_type{{0, 0, 0}}, point_type{{n0, n1, n2}},
-      tile_type{{4, 4, 4}}  // [TO DO] Choose optimal tile sizes for each device
-  );
-
-  Kokkos::parallel_for(
-      "KokkosFFT::crop_or_pad", range, KOKKOS_LAMBDA(int i0, int i1, int i2) {
-        out(i0, i1, i2) = in(i0, i1, i2);
-      });
-}
-
-template <typename ExecutionSpace, typename InViewType, typename OutViewType>
-void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-                      OutViewType& out, shape_type<4> s) {
-  constexpr std::size_t DIM = 4;
-
-  auto [s0, s1, s2, s3] = s;
-  out                   = OutViewType("out", s0, s1, s2, s3);
-
-  int n0 = std::min(s0, in.extent(0));
-  int n1 = std::min(s1, in.extent(1));
-  int n2 = std::min(s2, in.extent(2));
-  int n3 = std::min(s3, in.extent(3));
-
-  using range_type = Kokkos::MDRangePolicy<
-      ExecutionSpace,
-      Kokkos::Rank<DIM, Kokkos::Iterate::Default, Kokkos::Iterate::Default>>;
-  using tile_type  = typename range_type::tile_type;
-  using point_type = typename range_type::point_type;
-
-  range_type range(exec_space, point_type{{0, 0, 0, 0}},
-                   point_type{{n0, n1, n2, n3}}, tile_type{{4, 4, 4, 4}}
-                   // [TO DO] Choose optimal tile sizes for each device
-  );
-
-  Kokkos::parallel_for(
-      "KokkosFFT::crop_or_pad", range,
-      KOKKOS_LAMBDA(int i0, int i1, int i2, int i3) {
-        out(i0, i1, i2, i3) = in(i0, i1, i2, i3);
-      });
-}
-
-template <typename ExecutionSpace, typename InViewType, typename OutViewType>
-void crop_or_pad_impl(const ExecutionSpace& exec_space, const InViewType& in,
-                      OutViewType& out, shape_type<5> s) {
-  constexpr std::size_t DIM = 5;
+void crop_or_pad_ND(const ExecutionSpace& exec_space, const InViewType& in,
+                    const OutViewType& out) {
+  [[maybe_unused]] auto upper =
```

Comment:
Does the compiler still complain without `maybe_unused` ? To me it seems to be used in all branches.

Reply:
Maybe OK now. I will try to remove